added index.html

romantsarew opened the pull request:
+ modal window
+ catalog window in header

romantsarew committed:
added index.html

+ modal window
+ catalog window in header

Kartaviy-arteom commented on index.html:
```diff
@@ -1,12 +1,321 @@
+
   <head>
-    <meta charset="utf-8">
+    <meta charset="utf-8" />
```
meta не требует закрытия
подробнее тут https://css-live.ru/verstka/do-not-close-tags.html

romantsarew replied:
исправил

Kartaviy-arteom commented on index.html:
```diff
@@ -1,12 +1,321 @@
+        <a class="header-logo" href="index.html">
+          <!-- вставить свг лого -->
+        </a>
+        <div class="header-background">
```
похоже на лишний div
н уверен что он тебе будет нужен

romantsarew replied:
оставил, нужен оказался

Kartaviy-arteom commented on index.html:
```diff
@@ -1,12 +1,321 @@
+              <legend class="visually-hidden">
+                Форма поиска товара на сайте.
+              </legend>
+              <input class="search-field" id="search" type="search" />
```
legend это заголовок для fieldset
тут тебе нужен связанный с input label

romantsarew replied:
исправил

Kartaviy-arteom commented on index.html:
```diff
@@ -1,12 +1,321 @@
+              <button class="search-button">Найти</button>
+            </form>
+            <ul class="personal-list">
+              <li class="personal-item">
+                <button class="enter-button">Войти</button>
+              </li>
+              <li class="personal-item">
+                <button class="compare-button">Сравнить</button>
+              </li>
+              <li class="personal-item">
+                <button class="basket-button">Корзина</button>
+              </li>
```
список ссылок

romantsarew replied:
исправил

Kartaviy-arteom commented on index.html:
```diff
@@ -1,12 +1,321 @@
+            </ul>
+          </div>
+          <nav class="navigation-header">
+            <span class="products-catalog">Каталог товаров</span>
```
это ссылка на каталог

romantsarew replied:
исправил

Kartaviy-arteom commented on index.html:
```diff
@@ -1,12 +1,321 @@
+          </div>
+          <nav class="navigation-header">
+            <span class="products-catalog">Каталог товаров</span>
+            <button class="products-catalog-button" area-label="открыть окно с категориями товаров.">+</button>
```
+ - svg, aria-label/ хороший тон указывать кнопкам тип

Kartaviy-arteom commented on index.html:
```diff
@@ -1,12 +1,321 @@
+            </div>
+            <!-- добавить человечка бекграундом или псевдо -->
+          </div>
+          <a class="modal-button" href="#">
+            <img class="modal-picture" />
+            <button class="modal-text">Доставим редкий товар под заказ</button>
+            <button class="modal-info-button"></button>
+          </a>
```
кнопки внутри ссылки

Kartaviy-arteom commented on index.html:
```diff
@@ -1,12 +1,321 @@
+          </nav>
+        </div>
+      </header>
+      <section class="hot-offers">
```
это уже в main
последовательность заголовков: h2 выше h1

Kartaviy-arteom commented on index.html:
```diff
@@ -1,12 +1,321 @@
+        <nav class="navigation-products">
+          <ul class="navigation-products-list">
+            <li class="navigation-products-item">
+              <a class="navigation-products-link" href="#">
+                <!-- вставить свг -->
+                <span class="product-link-description">Виртуальная реальность</span>
+              </a>
+            </li>
+            <li class="navigation-products-item">
+              <a class="navigation-products-link" href="#">
+                <!-- вставить свг -->
+                <span class="product-link-description">Моноподы для селфи</span>
+              </a>
+            </li>
+            <li class="navigation-products-item">
+              <a class="navigation-products-link" href="#">
+                <!-- вставить свг -->
+                <span class="product-link-description">Экшн-камеры</span>
+              </a>
+            </li>
+            <li class="navigation-products-item">
+              <a class="navigation-products-link" href="#">
+                <!-- вставить свг -->
+                <span class="product-link-description">Фитнес-браслеты</span>
+              </a>
+            </li>
+            <li class="navigation-products-item">
+              <a class="navigation-products-link" href="#">
+                <!-- вставить свг -->
+                <span class="product-link-description">Умные часы</span>
+              </a>
+            </li>
+            <li class="navigation-products-item">
+              <a class="navigation-products-link" href="#">
+                <!-- вставить свг -->
+                <span class="product-link-description">квадрокоптеры</span>
+              </a>
+            </li>
+          </ul>
+        </nav>
```
лучше обойтись одним nav на странице

Kartaviy-arteom commented on index.html:
```diff
@@ -1,12 +1,321 @@
+        <h2 class="visually-hidden">Блок с горячими предложениями.</h2>
+        <div class="slider"></div>
+        <div class="offers-text">
+          <h2 class="second-title offers-title">
+            Порхает как бабочка, жалит как пчела!
+          </h2>
```
это не заголовок

Kartaviy-arteom commented on index.html:
```diff
@@ -1,12 +1,321 @@
+          </p>
+          <a class="offers-details" href="#">Подробнее</a>
+          <div class="offers-details-points">...</div>
+          <div class="offers-parametrs">
+            <strong class="main-text parameters-title">800 м</strong>
+            <span class="main-text parameters-text">Дальность полета</span>
+          </div>
+          <div class="offers-parametrs">
+            <strong class="main-text parameters-title">50 м</strong>
+            <span class="main-text parameters-text">Радиус поражения</span>
+          </div>
+        </div>
```
может быть таблицей, списком

Kartaviy-arteom commented on index.html:
```diff
@@ -1,12 +1,321 @@
+            замаскированным под стандартную камеру.
+          </p>
+          <a class="offers-details" href="#">Подробнее</a>
+          <div class="offers-details-points">...</div>
```
что это?

Kartaviy-arteom commented on index.html:
```diff
@@ -1,12 +1,321 @@
+          </ul>
+        </nav>
+        <section class="delivery-block">
+          <h2 class="visually-hidden">Блок о доставке.</h2>
```
это блок не только о доставке. Это часто называют табами. Ты слева выбираешь доставку, гарантия, кредит, а справа для них меняется текст. Можно сделать по-разному но ссылки не нужны

Kartaviy-arteom commented on index.html:
```diff
@@ -1,12 +1,321 @@
+          </a>
+        </section>
+        <section class="contact-block">
+          <h2 class="visually-hidden">Обратная связь.</h2>
+          <div class="about-block">
+            <h2 class="second title about-title">О нас</h2>
```
зачем здесь два h2?

Kartaviy-arteom commented on index.html:
```diff
@@ -1,12 +1,321 @@
+            </ul>
+            <a class="delivery-link about-link" href="#">Подробнее о нас</a>
+          </div>
+          <div class="contacts-block">
```
лучше section

Kartaviy-arteom commented on index.html:
```diff
@@ -1,12 +1,321 @@
+            <h3 class="third-title shop-title">Интернет-магазин «Девайс»</h3>
+            <p class="main-text about-description">
+              Санкт-Петербург, набережная реки Карповки, 5, литера П. <br />
+              +7 (812) 812-12-12
```
телефон ссылка

Kartaviy-arteom commented on index.html:
```diff
@@ -1,12 +1,321 @@
+            распаковками, а также акции, спецпредложения и вот это вот всё.
+          </p>
+          <form class="subscribe-form">
+            <legend class="visually-hidden" for="subscribe">
+              Подписка на рассылку.
+            </legend>
```
legend тут не валиден 
https://developer.mozilla.org/ru/docs/Learn/Forms/How_to_structure_a_web_form

Kartaviy-arteom commented on index.html:
```diff
@@ -1,12 +1,321 @@
+            <legend class="visually-hidden" for="subscribe">
+              Подписка на рассылку.
+            </legend>
+            <input class="subscribe-field" id="subscribe" type="email" placeholder="Ваш e-mail" required />
```
label

Kartaviy-arteom commented on index.html:
```diff
@@ -1,12 +1,321 @@
+            <!-- добавить человечка бекграундом или псевдо -->
+          </div>
+          <a class="modal-button" href="#">
+            <img class="modal-picture" />
```
декоративная графика

Kartaviy-arteom commented on index.html:
```diff
@@ -1,12 +1,321 @@
+            Санкт-Петербург, набережная реки Карповки, 5, литера П.
+          </p>
+          <!-- сделать псевдо с черточкой желтой -->
+          <nav class="navigation-footer">
+            <ul class="footer-list">
+              <li class="footer-item">
+                <a class="footer-link" href="#">Доставка</a>
+              </li>
+              <li class="footer-item">
+                <a class="footer-link" href="#">Гарантия</a>
+              </li>
+              <li class="footer-item">
+                <a class="footer-link" href="#">Контакты</a>
+              </li>
+            </ul>
+          </nav>
```
тебе не обязательно все ссылки объединять в nav

romantsarew replied:
исправил

Kartaviy-arteom commented on index.html:
```diff
@@ -1,12 +1,321 @@
+          <a class="footer-phone-call" href="tel:+78128121212">Тел.: +7 (812) 812-12-12</a>
+        </div>
+        <div class="footer-second-line">
+          <span class="divider"></span>
```
для этого не нужен элемент в разметке

romantsarew replied:
исправил

Kartaviy-arteom commented on index.html:
```diff
@@ -1,12 +1,321 @@
+              <label class="delivery-modal-title" for="count">Количество:</label>
+              <div class="tooltip-button" data-title="Постараемся найти, сколько нужно. Привезём сколько получится по требованиям
+              к провозу багажа."></div>
+              <div class="modal-delivery-button">
+                <button class="button-minus" type="button" value="уменьшить количество"></button>
+              </div>
```
зачем эту кнопу оборачивать в div?

romantsarew replied:
исправил

Kartaviy-arteom commented on index.html:
```diff
@@ -1,12 +1,321 @@
+              <div class="modal-delivery-button">
+                <button class="button-minus" type="button" value="уменьшить количество"></button>
+              </div>
+              <input class="field count-field" type="number" id="count" name="counts" placeholder="1" required/>
```
label

romantsarew replied:
исправил

Kartaviy-arteom commented on index.html:
```diff
@@ -1,12 +1,321 @@
+          <div class="footer-social">
+            <ul class="social-list">
+              <li class="social-item">
+                <a class="social-telegram" href="https://t.me/htmlacademy">
```
ссылкам на сторонние ресурсы лучше добавлять target="_blank"

romantsarew replied:
исправил

romantsarew committed:
added styles to index.html

1. styles
2. modal window without styles
3. further window with catalog with styles

Kartaviy-arteom commented on index.html:
```diff
@@ -1,12 +1,374 @@
     <meta charset="utf-8">
-    <title>HTML Academy: Девайс</title>
+    <title>Девайс</title>
+    <link rel="stylesheet" href="./styles/styles.css" />
```
при использовании сервера точка не обязательна вначале` href="./styles/styles.css"`, можно и так ` href="/styles/styles.css"`,
/ корень текущего диска;
./текущий каталог;

romantsarew replied:
исправил

Kartaviy-arteom commented on index.html:
```diff
@@ -1,12 +1,374 @@
+      <div class="section-container">
+        <header class="main-header">
+          <a class="header-logo" href="index.html">
+            <svg aria-hidden="true" focusable="false" width="145" height="37" fill="none"
+              xmlns="http://www.w3.org/2000/svg">
+              <path
+                d="M27.22.39v35.3h-7.6v-2.4c-1.7 1.9-4.1 3.1-7.51 3.1C5.5 36.39 0 30.59 0 23.09c0-7.5 5.5-13.3 12.1-13.3 3.41 0 5.91 1.2 7.51 3.1V.39h7.6Zm-7.6 22.7c0-3.8-2.5-6.2-6.01-6.2-3.5 0-6 2.4-6 6.2 0 3.8 2.5 6.2 6 6.2 3.5-.1 6-2.4 6-6.2ZM43.73 29.59c2.1 0 3.8-.8 4.8-1.9l6.1 3.5c-2.5 3.4-6.2 5.2-11 5.2-8.6 0-14.01-5.8-14.01-13.3 0-7.5 5.4-13.3 13.4-13.3 7.41 0 12.82 5.7 12.82 13.3 0 1.1-.1 ...
+                fill="currentColor" />
+              <path d="M83.46.39v10.1l7.6-10.1h-7.6Z" fill="#510094" />
+            </svg>
```
current color было в данном случае не указывать, у тебя в стайлгайде меняется только прозрачность

romantsarew replied:
исправил

Kartaviy-arteom commented on index.html:
```diff
@@ -1,12 +1,374 @@
-    <p>Репозиторий создан для обучения на профессиональном онлайн‑курсе «<a href="https://htmlacademy.ru/intensive/htmlcss">HTML и CSS. Профессиональная вёрстка сайтов</a>».</p>
 
+  <body>
+    <div class="page-container">
+      <div class="section-container">
```
тут я не понял для чего тебе именно два контейнера, почему не может быть один
на `section-container` добавлен display:grid, но для чего не понятно

romantsarew replied:
ну я специально сделал один контейнер для гридов, для хедера и секции со слайдером и описанием
так нельзя разве? а то переделывать там конечно кучу придется

Kartaviy-arteom commented on index.html:
```diff
@@ -1,12 +1,374 @@
+              <path d="M83.46.39v10.1l7.6-10.1h-7.6Z" fill="#510094" />
+            </svg>
+          </a>
+          <div class="header-background">
```
если этот элемент только для фона, он не нужен

romantsarew replied:
не только, на нем гриды висят, паддинги, отступы и наползание на логотип.

Kartaviy-arteom commented on index.html:
```diff
@@ -1,12 +1,374 @@
+            </svg>
+          </a>
+          <div class="header-background">
+            <div class="header-first-column">
```
лучше именовать классы от назначения, чем от внешнего вида. у тебя могут быть классы для примесей какич-то свойств. в препроцессорах для них добавятся mixin

romantsarew replied:
исправил

Kartaviy-arteom commented on index.html:
```diff
@@ -1,12 +1,374 @@
+                  Форма поиска товара на сайте.
+                </label>
+                <input class="search-field" id="search" type="search" placeholder="Текст поиска" required />
+                <button class="search-button">Найти</button>
```
type submit

Kartaviy-arteom commented on index.html:
```diff
@@ -1,12 +1,374 @@
+              </div>
+              <!-- конец выскакивающего окна -->
+            </div>
+            <nav class="header-second-column">
```
я бы сказал главная навигация на сайте: каталог, доставка, гарантия, контакты. Пользовательскую можно в nav не включать

Kartaviy-arteom commented on index.html:
```diff
@@ -1,12 +1,374 @@
+            <strong class="second-title promo-title">
+              Порхает как бабочка, жалит как пчела!
+            </strong>
+            <p class="main-text promo-description">
+              Этот обычный, на первый взгляд, квадрокоптер оснащен мощным лазером,
+              замаскированным под стандартную камеру.
+            </p>
+            <div class="promo-details">
+              <a class="button-usual promo-more" href="#">Подробнее</a>
+              <ol class="slider-pagination">
+                <li class="slider-item">
+                  <button class="slider-pagination-button slider-pagination-button-current" type="button">
+                    <span class="visually-hidden">1 фото</span>
+                  </button>
+                </li>
+                <li class="slider-item">
+                  <button class="slider-pagination-button" type="button">
+                    <span class="visually-hidden">2 фото</span>
+                  </button>
+                </li>
+                <li class="slider-item">
+                  <button class="slider-pagination-button" type="button">
+                    <span class="visually-hidden">3 фото</span>
+                  </button>
+                </li>
+              </ol>
+            </div>
+            <div class="promo-block">
+              <dl class="promo-parameters">
+                <dt class="main-text parameters-title">800 м</dt>
+                <dd class="main-text parameters-text">Дальность полета</dd>
+              </dl>
+              <dl class="promo-parameters">
+                <dt class="main-text parameters-title">50 м</dt>
+                <dd class="main-text parameters-text">Радиус поражения</dd>
+              </dl>
+            </div>
+          </div>
+        </section>
+      </div>
```
там у тебя 3 слайда. и там чуть другая разметка нужна. Есть пример слайдера в курсе

Kartaviy-arteom commented on index.html:
```diff
@@ -1,12 +1,374 @@
+        </section>
+      </div>
+      <main class="main-index">
+        <h1 class="visually-hidden">
```
у тебя есть критерий про иерархию заголовков, h2 не должен появляться раньше h1

Kartaviy-arteom commented on index.html:
```diff
@@ -1,12 +1,374 @@
+          <ul class="navigation-products-list">
+            <li class="navigation-products-item">
+              <a class="navigation-products-link" href="#">
+                <img class="navigation-image" src="./image/navigation/vr.svg" width="160" height="160"
+                  alt="виртуальная реальность.">
```
я бы сказал что это декоративные изображение, alt дублируешь подпись снизу https://up.htmlacademy.ru/htmlcss/34/criteries#ALLY

Kartaviy-arteom commented on index.html:
```diff
@@ -1,12 +1,374 @@
+                на ваш этаж.
+              </p>
+            </div>
+            <div class="visually-hidden guarantee-description description">
```
здесь тебе не нужно прятать их через visually-hidden

Kartaviy-arteom commented on index.html:
```diff
@@ -1,12 +1,374 @@
+            </div>
+          </div>
+        </section>
+        <section class="section-modal">
```
это может быть кнопка, но одна. да и секция с заголовком тут перебор

Kartaviy-arteom commented on index.html:
```diff
@@ -1,12 +1,374 @@
+          <button class="modal-button" type="button">Доставим редкий товар под заказ
+          </button>
+        </section>
+        <section class="contact-block">
+          <div class="about-block">
+            <h2 class="second-title about-title">О нас</h2>
+            <p class="main-text about-description">
+              Огромный выбор гаджетов не оставит равнодушным гика, который есть
+              в каждом из нас.
+            </p>
+            <p class="main-text about-description">
+              Мы можем доставить ваш товар в самые отдалённые уголки России
+              транспортными компаниями:
+            </p>
+            <ul class="delivery-companies-list">
+              <li class="delivery-companies-item">Меловые линии</li>
+              <li class="delivery-companies-item">СДИК</li>
+              <li class="delivery-companies-item">ТоксБерри</li>
+            </ul>
+            <a class="delivery-link" href="#">Подробнее о нас</a>
+          </div>
+          <section class="contacts-block">
+            <h2 class="second-title contacts-title">Контакты</h2>
+            <p class="main-text about-description">
+              Вы можете забрать товар сами, заехав в наш офис.<br> Заодно сможете
+              проверить работоспособность покупки.
+            </p>
+            <h3 class="third-title shop-title">Интернет-магазин «Девайс»</h3>
+            <p class="main-text about-description">
+              Санкт-Петербург, набережная реки Карповки, 5, литера П. <br />
+              <a class="phone-call" href="tel:+78128121212">+7 (812) 812-12-12</a>
+            </p>
+            <h3 class="third title timetable-title">Время работы</h3>
+            <p class="main-text timetable-description">
+              пн—пт: с 10:00 до 22:00 <br />сб—вс: с 10:00 до 19:00
+            </p>
+            <a class="contacts-link write-contanct" href="#">Напишите нам</a>
+          </section>
+        </section>
```
это див внутри которого две секции

Kartaviy-arteom commented on index.html:
```diff
@@ -1,12 +1,374 @@
+          <form class="subscribe-form" action="https://echo.htmlacademy.ru" method="post">
+            <label class="visually-hidden" for="subscribe">Форма подписки на рассылку</label>
+            <input class="subscribe-field" id="subscribe" type="email" placeholder="Ваш e-mail" required />
+            <button class="subscribe-button">Подписаться</button>
```
submit

Kartaviy-arteom commented on index.html:
```diff
@@ -1,12 +1,374 @@
+        </section>
+      </main>
+      <footer class="page-footer">
+        <div class="footer-first-column">
```
тебе не нужен класс, которые описывает номер колонки и далее

Kartaviy-arteom commented on styles/styles.css:
```diff
@@ -0,0 +1,1129 @@
+  padding: 15px 140px 0 140px;
+}
+
+.main-header {
+  grid-column: 1/-1;
+  max-width: 1160px;
+}
```
фон и паддинг верхний можно все перенести сюда.
Высота хедера на индексе и каталоге одинакова. сейчас он слишком большой

Kartaviy-arteom commented on styles/styles.css:
```diff
@@ -0,0 +1,1129 @@
+  z-index: 1;
+}
+
+.header-background {
+  position: relative;
+  display: grid;
+  grid-template-columns: 1fr 1fr;
+  padding: 45px 60px 135px 60px;
+  margin-top: -35px;
+  background-color: #ffe17f;
+  gap: 27px;
+  z-index: 0;
```
margin-top: -35px;
 и фон тут излишни

Kartaviy-arteom commented on styles/styles.css:
```diff
@@ -0,0 +1,1129 @@
+  z-index: 1;
+}
+
+.products-catalog::after {
+  content: "";
+  position: absolute;
+  background-image: url("../image/header/catalog-button.svg");
+  background-color: transparent;
+  border: none;
+  top: -17px;
+  left: 240px;
+  padding: 0;
+  cursor: pointer;
+  width: 50px;
+  height: 50px;
+  z-index: 1;
+}
```
часть не кликабельна, перекрывается другим элементом

Kartaviy-arteom commented on styles/styles.css:
```diff
@@ -0,0 +1,1129 @@
+  z-index: -1;
+}
+
+.slider-pagination {
```
будет позиционироваться абсолютно от низа слайдера

Kartaviy-arteom commented on styles/styles.css:
```diff
@@ -0,0 +1,1129 @@
+  max-width: 100%;
+}
+
+.footer-first-column {
```
если тебе нужны колонки  и строки, у тебя есть grid

Kartaviy-arteom commented on styles/styles.css:
```diff
@@ -0,0 +1,1129 @@
+}
+
+.services-item:last-child {
+  margin-bottom: 75px;
```
для чего это?

romantsarew replied:
чтобы снизу отступ был и весь блок имел высоту требуюмую

Kartaviy-arteom commented on styles/styles.css:
```diff
@@ -0,0 +1,1129 @@
+  position: absolute;
+  content: "";
+  width: 7px;
+  height: 319px;
```
высота должна зависить от высоты списка

romantsarew replied:
min-height? но не особо помогло

Kartaviy-arteom commented on styles/styles.css:
```diff
@@ -0,0 +1,1129 @@
+  list-style-type: none;
+}
+
+.header-button {
+  color: #000000;
+  text-decoration: none;
+  font-size: 18px;
+  line-height: 30px;
+}
```
размер активной области

Kartaviy-arteom commented on styles/styles.css:
```diff
@@ -0,0 +1,1129 @@
+}
+
+.search-button {
+  padding: 7px 53px;
```
тебе не нужно зажимать текст паддингами. задай кнопке ширину.

Kartaviy-arteom commented on styles/styles.css:
```diff
@@ -0,0 +1,1129 @@
+
+.header-background::after {
+  position: absolute;
+  content: "01";
```
сейчас тебе нужно для каждого слайда передавать в стили контент, будет проще оставить номер в разметке
